src/backend/columnar/write_state_management.c

```diff
@@ -176,7 +177,18 @@ columnar_init_write_state(Relation relation, TupleDesc tupdesc,
 	MemoryContext oldContext = MemoryContextSwitchTo(WriteStateContext);
 
 	ColumnarOptions columnarOptions = { 0 };
-	ReadColumnarOptions(relation->rd_id, &columnarOptions);
+
+	/* In case of an ALTER command, we set the original table's columnar options
+	 * to the new table here
+	 */
+	if (tupSlotRelationId != relation->rd_id)
+	{
+		ReadColumnarOptions(tupSlotRelationId, &columnarOptions);
+	}
+	else
+	{
+		ReadColumnarOptions(relation->rd_id, &columnarOptions);
+	}
```

Comment:
This if statement seems somewhat confusing to me, what we say here is that if `tupSlotRelationId` is not equal to `relation->rd_id` we read from `tupSlotRelationId`, if they are equal we read from `relation->rd_id`.

Doesn't this equate to the following:
```suggestion
	ReadColumnarOptions(tupSlotRelationId, &columnarOptions);
```

Meaning, don't we always simply read from the original relation oid?